Mathlib/Order/Filter/Cocardinal.lean

```diff
@@ -0,0 +1,70 @@
+/-
+Copyright (c) 2024 Josha Dekker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Josha Dekker
+-/
+import Mathlib.Order.Filter.CountableInter
+import Mathlib.SetTheory.Cardinal.Ordinal
+
+/-!
+# The cocardinal filter
+
+In this file we define `Filter.cocardinal hc`: the filter of sets with cardinality less than
+  `c` that satisfies `Cardinal.aleph0 < c`.
+
+-/
+
+open Set Filter
+
+universe u
+
+variable {α : Type*}
+
+namespace Filter
+
+/-- Construct a filter with cardinal intersection property. This constructor deduces
+`Filter.univ_sets` and `Filter.inter_sets` from the cardinal intersection property. -/
+def ofCardinalInter {c : Cardinal} (hc : Cardinal.aleph0 ≤ c) (l : Set (Set α))
+    (hp : ∀ S : Set (Set α), (Cardinal.mk S < c) → S ⊆ l → ⋂₀ S ∈ l)
+    (h_mono : ∀ s t, s ∈ l → s ⊆ t → t ∈ l) : Filter α where
+  sets := l
+  univ_sets := by
+    apply @sInter_empty α ▸ hp ∅ ?_ (empty_subset l)
+    rw [Cardinal.mk_eq_zero]
+    exact gt_of_ge_of_gt hc Cardinal.aleph0_pos
+  sets_of_superset := fun {x y} a a_1 ↦ h_mono x y a a_1
+  inter_sets {s t} hs ht := by
+    apply sInter_pair s t ▸ hp {s, t} ?_
+    · exact insert_subset_iff.2 ⟨hs, singleton_subset_iff.2 ht⟩
+    · exact gt_of_ge_of_gt hc (Cardinal.lt_aleph0_of_finite _)
+
+/-- Construct a filter sets whose complements satisfy a property that is stable under unions
+with a certain cardinality. -/
+def ofCardinalUnion {c : Cardinal} (hc : Cardinal.aleph0 ≤ c) (p : Set α → Prop)
+    (hUnion : ∀ S : Set (Set α), (Cardinal.mk S < c) → (∀ s ∈ S, p s) → p (⋃₀ S))
+    (hmono : ∀ t, p t → ∀ s ⊆ t, p s) : Filter α := by
+  refine .ofCardinalInter hc {s | p sᶜ} (fun S hSc hSp ↦ ?_)
+    fun s t ht hsub ↦ hmono sᶜ ht tᶜ (compl_subset_compl.2 hsub)
+  rw [mem_setOf_eq, compl_sInter]
+  exact hUnion (compl '' S) (lt_of_le_of_lt Cardinal.mk_image_le hSc) (ball_image_iff.2 hSp)
+
+-- Of course, this would generalise to CardinalInterFilter under a suitable definition for these.
+instance countableInter_ofCardinalnter {c : Cardinal} (hc : Cardinal.aleph0 < c) (l : Set (Set α))
```

Comment:
This can't be an `instance` because Lean can't guess `hc : aleph0 < c` from available `hc.le : aleph0 ≤ c`.

Reply:
But you only get a `CountableInterFilter` from this construction if `aleph0 < c`. Otherwise, we only have stability under finite intersections.

Reply:
Perhaps we may want `aleph0 < c` everywhere anyway, so that this explicitly does not nest the case for finite intersections?

Reply:
I've redefined some things, I think the problem should be gone now.